swatch-system-conduit/src/main/java/org/candlepin/subscriptions/conduit/inventory/ConduitFacts.java

```diff
@@ -76,4 +88,15 @@ public Integer getCpuCores() {
   public Long getMemory() {
     return super.getMemory();
   }
+
+  @Override
```

Comment:
This was to resolve a issue I had with Sonar qube, since I add a additional field in the ConduitFacts. The check failed as it wants subclass that have different fields than the parent class, to override the equal method.
So this is why I have that there, and some unit test were failing due to them expecting ConsumerInventory, but they got a different argument instead.

Reply:
@ksynvrit gotcha....I think the equals should compare network interfaces as well

Reply:
Yeah I tried before but it was cause some unit test to fail because of the the different arguments. So I left this as a work around. If you really think it would should do it, maybe we can tackle it together?

Reply:
I tried this locally too, I see why it happens (`null != new ArrayList<>()`), and actually I think the correct thing here to do is to leave the networkinterfaces as null if we have 0 NIC entries. Leaving a suggestion for where to do this.